gcloud-java-logging/src/main/java/com/google/gcloud/logging/spi/v2/testing/LocalLoggingImpl.java

```diff
@@ -0,0 +1,97 @@
+package com.google.gcloud.pubsub.spi.v2.testing;
+
+import com.google.logging.v2.DeleteLogRequest;
+import com.google.logging.v2.ListLogEntriesRequest;
+import com.google.logging.v2.ListLogEntriesResponse;
+import com.google.logging.v2.ListMonitoredResourceDescriptorsRequest;
+import com.google.logging.v2.ListMonitoredResourceDescriptorsResponse;
+import com.google.logging.v2.LogEntry;
+import com.google.logging.v2.LoggingServiceV2Grpc;
+import com.google.logging.v2.ReadLogEntriesRequest;
+import com.google.logging.v2.ReadLogEntriesResponse;
+import com.google.logging.v2.WriteLogEntriesRequest;
+import com.google.logging.v2.WriteLogEntriesResponse;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Empty;
+
+import io.grpc.stub.StreamObserver;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class LocalLoggingImpl implements LoggingServiceV2Grpc.LoggingServiceV2 {
+
+  private Map<String, List<LogEntry>> logs = new HashMap<>();
+
+  @Override
+  public void deleteLog(DeleteLogRequest request, StreamObserver<Empty> responseObserver) {
+    logs.remove(request.getLogName());
+    responseObserver.onNext(Empty.getDefaultInstance());
+    responseObserver.onCompleted();
+  }
+
+  @Override
+  public void writeLogEntries(WriteLogEntriesRequest request,
+      StreamObserver<WriteLogEntriesResponse> responseObserver) {
+    List<LogEntry> entries = logs.get(request.getLogName());
+    if (entries == null) {
```

Comment:
I was not expecting to see this pattern in the API yet. I suggest we turn off the StreamObersver pattern for now. We can discuss bringing it back when the design and implementation pattern are finalized, including making it thread safe.